## Mapeo de columnas

Este cuaderno carga el archivo Excel de evaluacion y crea una tabla con la descripcion de cada columna de la hoja 'Base' usando la informacion de la hoja 'Diccionario'.

In [1]:
from pathlib import Path

import pandas as pd

In [2]:
DATA_PATH = Path("Base_Evaluacion_oct_2024 (2) (1).xlsx")
DATA_PATH.exists()

True

In [3]:
# Cargar metadatos del libro
xl = pd.ExcelFile(DATA_PATH)

base_columns = (
    pd.read_excel(DATA_PATH, sheet_name="Base", nrows=0)
    .columns
    .str.strip()
)

dictionary_raw = pd.read_excel(DATA_PATH, sheet_name="Diccionario")
dictionary = dictionary_raw.copy()
dictionary.columns = dictionary.columns.str.strip()
dictionary["columna_diccionario"] = dictionary["DESCRIPCION CDA BATCH"].str.strip()
dictionary["columna_base"] = dictionary["columna_diccionario"].str.lower()

column_metadata = (
    pd.DataFrame({"columna_base": base_columns})
    .merge(
        dictionary[
            [
                "columna_base",
                "DESCRIPCION CDA WEB",
                "columna_diccionario",
                "TIPO",
                "LONGTUD MAX",
            ]
        ],
        on="columna_base",
        how="left",
    )
    .rename(
        columns={
            "DESCRIPCION CDA WEB": "descripcion_web",
            "columna_diccionario": "nombre_diccionario",
            "TIPO": "tipo",
            "LONGTUD MAX": "longitud_max",
        }
    )
)

column_metadata

columna_base            descripcion_web nombre_diccionario        tipo  \
0              ID                        NaN                NaN         NaN   
1    cal_nom_afip           % Calidad Nombre       CAL_NOM_AFIP  Porcentaje   
2            edad                       Edad               EDAD      Entero   
3        lc_monto               Monto Límite           LC_MONTO      Moneda   
4      lc_monto_c         Monto Límite Cuota         LC_MONTO_C      Moneda   
..            ...                        ...                ...         ...   
99       bs_posee  Posee Beneficios Sociales           BS_POSEE    Booleano   
100   com_ext_imp              Es Importador        COM_EXT_IMP    Booleano   
101   com_ext_exp              Es Exportador        COM_EXT_EXP    Booleano   
102  ft_apocrifas         Posee FT Apócrifas       FT_APOCRIFAS    Booleano   
103        target                        NaN                NaN         NaN   

     longitud_max  
0             NaN  
1             3.0  
2             3.0  
3            20.0  
4            20.0  
..            ...  
99            2.0  
100           2.0  
101           2.0  
102           2.0  
103           NaN  

[104 rows x 5 columns]

Las columnas que no cuentan con descripcion en la hoja 'Diccionario' se listan a continuacion (si el resultado esta vacio, todas las columnas tienen descripcion).

In [4]:
column_metadata[column_metadata["descripcion_web"].isna()]

columna_base descripcion_web nombre_diccionario tipo  longitud_max
0               ID             NaN                NaN  NaN           NaN
33      rat_cm_nse             NaN                NaN  NaN           NaN
61      descub_mto             NaN                NaN  NaN           NaN
68     rc_vg_otras             NaN                NaN  NaN           NaN
75    rc_vg_telcos             NaN                NaN  NaN           NaN
87   cod_ganancias             NaN                NaN  NaN           NaN
103         target             NaN                NaN  NaN           NaN

### Columnas del dataset y significado

Total de columnas en la hoja : 104.
Cada item incluye nombre de la columna en la base, la etiqueta en el diccionario y los metadatos disponibles.

- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : % Calidad Nombre (diccionario: , tipo: Porcentaje, longitud_max: 3)
- : Edad (diccionario: , tipo: Entero, longitud_max: 3)
- : Monto Límite (diccionario: , tipo: Moneda, longitud_max: 20)
- : Monto Límite Cuota (diccionario: , tipo: Moneda, longitud_max: 20)
- : Score Servicio (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Servicio (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Servicio (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Servicio (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Servicio Tendencia (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 20)
- : Score Servicio Tendencia (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 20)
- : Score Servicio Tendencia (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 20)
- : Score Financiero (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Financiero (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Financiero (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Financiero (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 3)
- : Score Financiero Tendencia (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 20)
- : Score Financiero Tendencia (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 20)
- : Score Financiero Tendencia (-12 Ms) (diccionario: , tipo: Texto, longitud_max: 8)
- : Telcos Consultadas - Cantidad (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : Telcos Morosas - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : Telcos Morosas - Deuda Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : Telcos Morosas - Antigüedad Max. Deuda (diccionario: , tipo: Entero, longitud_max: 4)
- : Telcos Morosas - Cantidad (Sin G. Telecom) (diccionario: , tipo: Entero, longitud_max: 6)
- : Deuda Total - Movistar (diccionario: , tipo: Moneda, longitud_max: 20)
- : Deuda Total - Telefónica (diccionario: , tipo: Moneda, longitud_max: 20)
- : Deuda Total - Claro (diccionario: , tipo: Moneda, longitud_max: 20)
- : Deuda Total - Personal (diccionario: , tipo: Moneda, longitud_max: 20)
- : Deuda Total - Telecom (diccionario: , tipo: Moneda, longitud_max: 20)
- : NSE Percentil (diccionario: , tipo: Entero, longitud_max: 3)
- : FE (diccionario: , tipo: Entero, longitud_max: 2)
- : FE Aperturada (diccionario: , tipo: Entero, longitud_max: 2)
- : Compromisos Mensuales (diccionario: , tipo: Moneda, longitud_max: 20)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : Productos Financieros Activos (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : BCRA Antigüedad (Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : BCRA Situación (diccionario: , tipo: Entero, longitud_max: 2)
- : BCRA Peor Situación (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 2)
- : BCRA Peor Situación (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 2)
- : BCRA Peor Situación (-12 Ms) (diccionario: , tipo: Texto, longitud_max: 8)
- : BCRA Tendencia Situación (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 2)
- : BCRA Tendencia Situación (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 2)
- : BCRA Tendencia Situación (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 2)
- : Bancarizado (diccionario: , tipo: Booleano, longitud_max: 2)
- : Bancos Cantidad (diccionario: , tipo: Entero, longitud_max: 2)
- : Bancos Endeudamiento Vigente (diccionario: , tipo: Moneda, longitud_max: 20)
- : TC Cantidad (diccionario: , tipo: Entero, longitud_max: 2)
- : TC Saldo Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : Cheques Sin Fondos - Cantidad (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : Cheques Sin Fondos - Monto (-3 Ms) (diccionario: , tipo: Moneda, longitud_max: 20)
- : Cheques Sin Fondos - Cantidad (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : Cheques Sin Fondos - Monto (-6 Ms) (diccionario: , tipo: Moneda, longitud_max: 20)
- : P. Prendarios - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : P. Prendarios - Saldo Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : P. Hipotecarios - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : P. Hipotecarios - Saldo Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : P. Personales - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : P. Personales - Saldo Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : P. Otros - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : P. Otros - Saldo Total (diccionario: , tipo: Moneda, longitud_max: 20)
- : Líneas de Descubierto vigentes - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : RC Vigentes - Cantidad - (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : RC Vigentes - Cantidad - (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : RC Vigentes - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : RC - Fuente CAP1 (diccionario: , tipo: Entero, longitud_max: 6)
- : RC - Fuente CAP2 (diccionario: , tipo: Entero, longitud_max: 6)
- : RC - Fuente CAP3 (diccionario: , tipo: Entero, longitud_max: 6)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : RF Vigentes - Cantidad - (-6 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : RF Vigentes - Cantidad - (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : RF Vigentes - Cantidad (diccionario: , tipo: Entero, longitud_max: 6)
- : RF - Fuente CAP1 (diccionario: , tipo: Entero, longitud_max: 6)
- : RF - Fuente CAP2 (diccionario: , tipo: Entero, longitud_max: 6)
- : RF - Fuente CAP3 (diccionario: , tipo: Entero, longitud_max: 6)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : Es Empleado? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Jubilado? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Pensionado? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Monotributista? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Autónomo? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Empleador? (diccionario: , tipo: Entero, longitud_max: 2)
- : Es Integrante Sociedad? (diccionario: , tipo: Entero, longitud_max: 2)
- : Sin Actividad? (diccionario: , tipo: Entero, longitud_max: 2)
- : Antigüedad en AFIP (diccionario: , tipo: Entero, longitud_max: 6)
- : Antigüedad Laboral Max - Empleador Vigente (diccionario: , tipo: Entero, longitud_max: 6)
- : Cantidad Empleados (diccionario: , tipo: Entero, longitud_max: 6)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)
- : APE Estado (Peor Situación) (diccionario: , tipo: Entero, longitud_max: 6)
- : APE Impagos - Cantidad (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APE Impagos - Cantidad (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APE Parcial - Cantidad (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APE Empleadores - Cantidad (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APER Estado (Peor Situación) (diccionario: , tipo: Entero, longitud_max: 6)
- : APER Impagos - Cantidad (-3 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APER Impagos - Cantidad (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : APER Parcial - Cantidad (-12 Ms) (diccionario: , tipo: Entero, longitud_max: 6)
- : Falta de pago (diccionario: , tipo: Booleano, longitud_max: 2)
- : Código obra social (diccionario: , tipo: Entero, longitud_max: 6)
- : Posee Beneficios Sociales (diccionario: , tipo: Booleano, longitud_max: 2)
- : Es Importador (diccionario: , tipo: Booleano, longitud_max: 2)
- : Es Exportador (diccionario: , tipo: Booleano, longitud_max: 2)
- : Posee FT Apócrifas (diccionario: , tipo: Booleano, longitud_max: 2)
- : Descripción pendiente en diccionario (diccionario: , tipo: N/D, longitud_max: N/D)

### Guía para modelar la columna 

- **Objetivo**: la columna  indica si la persona incurre en default; trataremos de predecirla como variable binaria.
- **Variables sociodemográficas**: , , , , ,  permiten caracterizar perfil y estabilidad laboral.
- **Historial crediticio**: puntajes , , situaciones , montos de préstamos (, ), cuotas () capturan comportamiento financiero previo y capacidad de pago.
- **Comportamiento en telecom**: indicadores , ,  aportan señales sobre morosidad en servicios y relacion con la compañía.
- **Actividad económica y fiscal**: banderas , , , registros AFIP (, ) ayudan a segmentar flujo de ingresos y cumplimiento fiscal.
- **Construcción del modelo**: limpiar valores faltantes, normalizar montos, crear variables derivadas (ratios, tendencias de puntajes), codificar categorías y evaluar importancia (árboles de decisión, gradient boosting, modelos lineales regularizados).
- **Evaluación y fairness**: usar validación temporal si hay fechas, métricas como AUC/Recall, y monitorear sesgos en segmentos (edad, género, zona).
- **Refinamiento**: seleccionar variables con información en diccionario, explorar interacciones (ej.  vs ), y documentar columnas sin descripción para completar metadatos antes de producción.

### Columnas del dataset y significado

Total de columnas en la hoja `Base`: 104.
Cada item incluye el nombre de la columna en la base, la etiqueta en el diccionario y los metadatos disponibles.

- `ID`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `cal_nom_afip`: % Calidad Nombre (diccionario: `CAL_NOM_AFIP`, tipo: Porcentaje, longitud_max: 3)
- `edad`: Edad (diccionario: `EDAD`, tipo: Entero, longitud_max: 3)
- `lc_monto`: Monto Límite (diccionario: `LC_MONTO`, tipo: Moneda, longitud_max: 20)
- `lc_monto_c`: Monto Límite Cuota (diccionario: `LC_MONTO_C`, tipo: Moneda, longitud_max: 20)
- `sco_ser`: Score Servicio (diccionario: `SCO_SER`, tipo: Entero, longitud_max: 3)
- `sco_ser_3m`: Score Servicio (-3 Ms) (diccionario: `SCO_SER_3M`, tipo: Entero, longitud_max: 3)
- `sco_ser_6m`: Score Servicio (-6 Ms) (diccionario: `SCO_SER_6M`, tipo: Entero, longitud_max: 3)
- `sco_ser_12m`: Score Servicio (-12 Ms) (diccionario: `SCO_SER_12M`, tipo: Entero, longitud_max: 3)
- `sco_ser_3m_t`: Score Servicio Tendencia (-3 Ms) (diccionario: `SCO_SER_3M_T`, tipo: Entero, longitud_max: 20)
- `sco_ser_6m_t`: Score Servicio Tendencia (-6 Ms) (diccionario: `SCO_SER_6M_T`, tipo: Entero, longitud_max: 20)
- `sco_ser_12m_t`: Score Servicio Tendencia (-12 Ms) (diccionario: `SCO_SER_12M_T`, tipo: Entero, longitud_max: 20)
- `sco_fin`: Score Financiero (diccionario: `SCO_FIN`, tipo: Entero, longitud_max: 3)
- `sco_fin_3m`: Score Financiero (-3 Ms) (diccionario: `SCO_FIN_3M`, tipo: Entero, longitud_max: 3)
- `sco_fin_6m`: Score Financiero (-6 Ms) (diccionario: `SCO_FIN_6M`, tipo: Entero, longitud_max: 3)
- `sco_fin_12m`: Score Financiero (-12 Ms) (diccionario: `SCO_FIN_12M`, tipo: Entero, longitud_max: 3)
- `sco_fin_3m_t`: Score Financiero Tendencia (-3 Ms) (diccionario: `SCO_FIN_3M_T`, tipo: Entero, longitud_max: 20)
- `sco_fin_6m_t`: Score Financiero Tendencia (-6 Ms) (diccionario: `SCO_FIN_6M_T`, tipo: Entero, longitud_max: 20)
- `sco_fin_12m_t`: Score Financiero Tendencia (-12 Ms) (diccionario: `SCO_FIN_12M_T`, tipo: Texto, longitud_max: 8)
- `telcos_cons_3m`: Telcos Consultadas - Cantidad (-3 Ms) (diccionario: `TELCOS_CONS_3M`, tipo: Entero, longitud_max: 6)
- `telcos_mor_cant`: Telcos Morosas - Cantidad (diccionario: `TELCOS_MOR_CANT`, tipo: Entero, longitud_max: 6)
- `telcos_mor_dda`: Telcos Morosas - Deuda Total (diccionario: `TELCOS_MOR_DDA`, tipo: Moneda, longitud_max: 20)
- `telcos_mor_ant`: Telcos Morosas - Antigüedad Max. Deuda (diccionario: `TELCOS_MOR_ANT`, tipo: Entero, longitud_max: 4)
- `telcos_mor_cant2`: Telcos Morosas - Cantidad (Sin G. Telecom) (diccionario: `TELCOS_MOR_CANT2`, tipo: Entero, longitud_max: 6)
- `telcos_dda_mv`: Deuda Total - Movistar (diccionario: `TELCOS_DDA_MV`, tipo: Moneda, longitud_max: 20)
- `telcos_dda_tf`: Deuda Total - Telefónica (diccionario: `TELCOS_DDA_TF`, tipo: Moneda, longitud_max: 20)
- `telcos_dda_cl`: Deuda Total - Claro (diccionario: `TELCOS_DDA_CL`, tipo: Moneda, longitud_max: 20)
- `telcos_dda_tp`: Deuda Total - Personal (diccionario: `TELCOS_DDA_TP`, tipo: Moneda, longitud_max: 20)
- `telcos_dda_ta`: Deuda Total - Telecom (diccionario: `TELCOS_DDA_TA`, tipo: Moneda, longitud_max: 20)
- `nse_perc`: NSE Percentil (diccionario: `NSE_PERC`, tipo: Entero, longitud_max: 3)
- `fe`: FE (diccionario: `FE`, tipo: Entero, longitud_max: 2)
- `fe_max`: FE Aperturada (diccionario: `FE_MAX`, tipo: Entero, longitud_max: 2)
- `compr_mens`: Compromisos Mensuales (diccionario: `COMPR_MENS`, tipo: Moneda, longitud_max: 20)
- `rat_cm_nse`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `pfina_12m`: Productos Financieros Activos (-12 Ms) (diccionario: `PFINA_12M`, tipo: Entero, longitud_max: 6)
- `bcra_ant`: BCRA Antigüedad (Ms) (diccionario: `BCRA_ANT`, tipo: Entero, longitud_max: 6)
- `bcra_sit_vg`: BCRA Situación (diccionario: `BCRA_SIT_VG`, tipo: Entero, longitud_max: 2)
- `bcra_psit_3m`: BCRA Peor Situación (-3 Ms) (diccionario: `BCRA_PSIT_3M`, tipo: Entero, longitud_max: 2)
- `bcra_psit_6m`: BCRA Peor Situación (-6 Ms) (diccionario: `BCRA_PSIT_6M`, tipo: Entero, longitud_max: 2)
- `bcra_psit_12m`: BCRA Peor Situación (-12 Ms) (diccionario: `BCRA_PSIT_12M`, tipo: Texto, longitud_max: 8)
- `bcra_psit_3m_t`: BCRA Tendencia Situación (-3 Ms) (diccionario: `BCRA_PSIT_3M_T`, tipo: Entero, longitud_max: 2)
- `bcra_psit_6m_t`: BCRA Tendencia Situación (-6 Ms) (diccionario: `BCRA_PSIT_6M_T`, tipo: Entero, longitud_max: 2)
- `bcra_psit_12m_t`: BCRA Tendencia Situación (-12 Ms) (diccionario: `BCRA_PSIT_12M_T`, tipo: Entero, longitud_max: 2)
- `bancrzdo`: Bancarizado (diccionario: `BANCRZDO`, tipo: Booleano, longitud_max: 2)
- `bco_cant`: Bancos Cantidad (diccionario: `BCO_CANT`, tipo: Entero, longitud_max: 2)
- `bco_endeu`: Bancos Endeudamiento Vigente (diccionario: `BCO_ENDEU`, tipo: Moneda, longitud_max: 20)
- `tc_cant`: TC Cantidad (diccionario: `TC_CANT`, tipo: Entero, longitud_max: 2)
- `tc_sdot`: TC Saldo Total (diccionario: `TC_SDOT`, tipo: Moneda, longitud_max: 20)
- `che_sf_cant_3m`: Cheques Sin Fondos - Cantidad (-3 Ms) (diccionario: `CHE_SF_CANT_3M`, tipo: Entero, longitud_max: 6)
- `che_sf_mto_3m`: Cheques Sin Fondos - Monto (-3 Ms) (diccionario: `CHE_SF_MTO_3M`, tipo: Moneda, longitud_max: 20)
- `che_sf_cant_6m`: Cheques Sin Fondos - Cantidad (-6 Ms) (diccionario: `CHE_SF_CANT_6M`, tipo: Entero, longitud_max: 6)
- `che_sf_mto_6m`: Cheques Sin Fondos - Monto (-6 Ms) (diccionario: `CHE_SF_MTO_6M`, tipo: Moneda, longitud_max: 20)
- `pre_pre_cant`: P. Prendarios - Cantidad (diccionario: `PRE_PRE_CANT`, tipo: Entero, longitud_max: 6)
- `pre_pre_sldo`: P. Prendarios - Saldo Total (diccionario: `PRE_PRE_SLDO`, tipo: Moneda, longitud_max: 20)
- `pre_hip_cant`: P. Hipotecarios - Cantidad (diccionario: `PRE_HIP_CANT`, tipo: Entero, longitud_max: 6)
- `pre_hip_sldo`: P. Hipotecarios - Saldo Total (diccionario: `PRE_HIP_SLDO`, tipo: Moneda, longitud_max: 20)
- `pre_per_cant`: P. Personales - Cantidad (diccionario: `PRE_PER_CANT`, tipo: Entero, longitud_max: 6)
- `pre_per_sldo`: P. Personales - Saldo Total (diccionario: `PRE_PER_SLDO`, tipo: Moneda, longitud_max: 20)
- `pre_otr_cant`: P. Otros - Cantidad (diccionario: `PRE_OTR_CANT`, tipo: Entero, longitud_max: 6)
- `pre_otr_sldo`: P. Otros - Saldo Total (diccionario: `PRE_OTR_SLDO`, tipo: Moneda, longitud_max: 20)
- `descub_cant`: Líneas de Descubierto vigentes - Cantidad (diccionario: `DESCUB_CANT`, tipo: Entero, longitud_max: 6)
- `descub_mto`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `rc_vg_6m`: RC Vigentes - Cantidad - (-6 Ms) (diccionario: `RC_VG_6M`, tipo: Entero, longitud_max: 6)
- `rc_vg_3m`: RC Vigentes - Cantidad - (-3 Ms) (diccionario: `RC_VG_3M`, tipo: Entero, longitud_max: 6)
- `rc_vg`: RC Vigentes - Cantidad (diccionario: `RC_VG`, tipo: Entero, longitud_max: 6)
- `rc_vg_fcap1`: RC - Fuente CAP1 (diccionario: `RC_VG_FCAP1`, tipo: Entero, longitud_max: 6)
- `rc_vg_fcap2`: RC - Fuente CAP2 (diccionario: `RC_VG_FCAP2`, tipo: Entero, longitud_max: 6)
- `rc_vg_fcap3`: RC - Fuente CAP3 (diccionario: `RC_VG_FCAP3`, tipo: Entero, longitud_max: 6)
- `rc_vg_otras`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `rf_vg_6m`: RF Vigentes - Cantidad - (-6 Ms) (diccionario: `RF_VG_6M`, tipo: Entero, longitud_max: 6)
- `rf_vg_3m`: RF Vigentes - Cantidad - (-3 Ms) (diccionario: `RF_VG_3M`, tipo: Entero, longitud_max: 6)
- `rf_vg`: RF Vigentes - Cantidad (diccionario: `RF_VG`, tipo: Entero, longitud_max: 6)
- `rf_vg_fcap1`: RF - Fuente CAP1 (diccionario: `RF_VG_FCAP1`, tipo: Entero, longitud_max: 6)
- `rf_vg_fcap2`: RF - Fuente CAP2 (diccionario: `RF_VG_FCAP2`, tipo: Entero, longitud_max: 6)
- `rf_vg_fcap3`: RF - Fuente CAP3 (diccionario: `RF_VG_FCAP3`, tipo: Entero, longitud_max: 6)
- `rc_vg_telcos`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `es_empleado`: Es Empleado? (diccionario: `ES_EMPLEADO`, tipo: Entero, longitud_max: 2)
- `es_jubilado`: Es Jubilado? (diccionario: `ES_JUBILADO`, tipo: Entero, longitud_max: 2)
- `es_pensionado`: Es Pensionado? (diccionario: `ES_PENSIONADO`, tipo: Entero, longitud_max: 2)
- `es_monotrib`: Es Monotributista? (diccionario: `ES_MONOTRIB`, tipo: Entero, longitud_max: 2)
- `es_autonomo`: Es Autónomo? (diccionario: `ES_AUTONOMO`, tipo: Entero, longitud_max: 2)
- `es_empleador`: Es Empleador? (diccionario: `ES_EMPLEADOR`, tipo: Entero, longitud_max: 2)
- `es_integ_soc`: Es Integrante Sociedad? (diccionario: `ES_INTEG_SOC`, tipo: Entero, longitud_max: 2)
- `es_sin_activ`: Sin Actividad? (diccionario: `ES_SIN_ACTIV`, tipo: Entero, longitud_max: 2)
- `antig_afip`: Antigüedad en AFIP (diccionario: `ANTIG_AFIP`, tipo: Entero, longitud_max: 6)
- `antig_laboral_max`: Antigüedad Laboral Max - Empleador Vigente (diccionario: `ANTIG_LABORAL_MAX`, tipo: Entero, longitud_max: 6)
- `cant_empleados`: Cantidad Empleados (diccionario: `CANT_EMPLEADOS`, tipo: Entero, longitud_max: 6)
- `cod_ganancias`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)
- `ape_psit`: APE Estado (Peor Situación) (diccionario: `APE_PSIT`, tipo: Entero, longitud_max: 6)
- `ape_cant_imp_3m`: APE Impagos - Cantidad (-3 Ms) (diccionario: `APE_CANT_IMP_3M`, tipo: Entero, longitud_max: 6)
- `ape_cant_imp_12m`: APE Impagos - Cantidad (-12 Ms) (diccionario: `APE_CANT_IMP_12M`, tipo: Entero, longitud_max: 6)
- `ape_cant_parc_12m`: APE Parcial - Cantidad (-12 Ms) (diccionario: `APE_CANT_PARC_12M`, tipo: Entero, longitud_max: 6)
- `ape_cant_er_12m`: APE Empleadores - Cantidad (-12 Ms) (diccionario: `APE_CANT_ER_12M`, tipo: Entero, longitud_max: 6)
- `aper_psit`: APER Estado (Peor Situación) (diccionario: `APER_PSIT`, tipo: Entero, longitud_max: 6)
- `aper_cant_imp_3m`: APER Impagos - Cantidad (-3 Ms) (diccionario: `APER_CANT_IMP_3M`, tipo: Entero, longitud_max: 6)
- `aper_cant_imp_12m`: APER Impagos - Cantidad (-12 Ms) (diccionario: `APER_CANT_IMP_12M`, tipo: Entero, longitud_max: 6)
- `aper_cant_parc_12m`: APER Parcial - Cantidad (-12 Ms) (diccionario: `APER_CANT_PARC_12M`, tipo: Entero, longitud_max: 6)
- `art_fpgo`: Falta de pago (diccionario: `ART_FPGO`, tipo: Booleano, longitud_max: 2)
- `sss_os_cod`: Código obra social (diccionario: `SSS_OS_COD`, tipo: Entero, longitud_max: 6)
- `bs_posee`: Posee Beneficios Sociales (diccionario: `BS_POSEE`, tipo: Booleano, longitud_max: 2)
- `com_ext_imp`: Es Importador (diccionario: `COM_EXT_IMP`, tipo: Booleano, longitud_max: 2)
- `com_ext_exp`: Es Exportador (diccionario: `COM_EXT_EXP`, tipo: Booleano, longitud_max: 2)
- `ft_apocrifas`: Posee FT Apócrifas (diccionario: `FT_APOCRIFAS`, tipo: Booleano, longitud_max: 2)
- `target`: Descripcion pendiente en diccionario (diccionario: `N/D`, tipo: N/D, longitud_max: N/D)

### Guia para modelar la columna 

- **Objetivo**: la columna `target` indica si la persona incurre en default; se modela como variable binaria.
- **Variables sociodemograficas**: `edad`, `genero`, `nse_perc`, `antig_afip`, `antig_laboral_max`, `cant_empleados` caracterizan perfil y estabilidad laboral.
- **Historial crediticio**: `sco_ser*`, `sco_fin*`, indicadores `bcra_*`, prestamos `pre_*`, `descub_*`, limites `lc_monto*` reflejan comportamiento financiero y riesgo de mora.
- **Comportamiento en telecom**: `telcos_*`, `rc_vg_telcos`, `rc_vg_*` aportan senales de uso y morosidad en servicios de la compania.
- **Actividad economica y fiscal**: banderas `es_empleado`, `es_monotrib`, `cod_ganancias`, registros AFIP (`ape_*`, `aper_*`) ayudan a segmentar capacidad de ingreso y cumplimiento fiscal.
- **Construccion del modelo**: limpiar faltantes, escalar montos, crear ratios o tendencias, codificar categorias y evaluar modelos (logistica regularizada, gradient boosting, random forest).
- **Evaluacion y fairness**: usar validacion temporal si aplica, medir AUC/recall y monitorear sesgos por segmentos (edad, genero, nivel socioeconomico).
- **Refinamiento**: priorizar variables con respaldo en el diccionario, explorar interacciones (`sco_fin` con `bcra_psit_*`), y documentar campos sin descripcion antes de deploy.